things/mocks/channels.go

```diff
@@ -114,20 +114,20 @@ func (crm *channelRepositoryMock) RetrieveAll(_ context.Context, owner string, p
 	return page, nil
 }
 
-func (crm *channelRepositoryMock) RetrieveByThing(_ context.Context, owner, thingID string, offset, limit uint64, connected bool) (things.ChannelsPage, error) {
+func (crm *channelRepositoryMock) RetrieveByThing(_ context.Context, owner, thID string, pm things.PageMetadata) (things.ChannelsPage, error) {
 	channels := make([]things.Channel, 0)
 
-	if offset < 0 || limit <= 0 {
+	if pm.Offset < 0 || pm.Limit <= 0 {
 		return things.ChannelsPage{}, nil
 	}
 
-	first := uint64(offset) + 1
-	last := first + uint64(limit)
+	first := uint64(pm.Offset) + 1
+	last := first + uint64(pm.Limit)
 
 	// Append connected or not connected channels
-	switch connected {
+	switch pm.Connected {
 	case true:
-		for _, co := range crm.cconns[thingID] {
+		for _, co := range crm.cconns[thID] {
```

Comment:
what is this co, why are you making index out of its id
and I'm confused since
`cconns   map[string]map[string]things.Channel`
than
`cconns[thID]` should be `map[string]things.Channel`

Reply:
indeed, strange